pkg/render/intrusion_detection.go

```diff
@@ -206,41 +206,48 @@ func (c *intrusionDetectionComponent) Objects() ([]client.Object, []client.Objec
 
 	// AD Related deployment only for management/standalone cluster
 	// When FIPS mode is enabled, we currently disable our python based images.
-	if !c.cfg.ManagedCluster && !operatorv1.IsFIPSModeEnabled(c.cfg.Installation.FIPSMode) {
-		// Service + Deployment + RBAC for AD API
-		objs = append(objs,
-			c.adAPIAllowTigeraPolicy(),
-			c.adAPIServiceAccount(),
-			c.adAPIAccessClusterRole(),
-			c.adAPIAccessRoleBinding(),
-		)
+	if !c.cfg.ManagedCluster {
+		if !operatorv1.IsFIPSModeEnabled(c.cfg.Installation.FIPSMode) {
+			// Service + Deployment + RBAC for AD API
+			objs = append(objs,
+				c.adAPIAllowTigeraPolicy(),
+				c.adAPIServiceAccount(),
+				c.adAPIAccessClusterRole(),
+				c.adAPIAccessRoleBinding(),
+			)
 
-		shouldConfigureADStorage := len(c.cfg.IntrusionDetection.Spec.AnomalyDetection.StorageClassName) > 0
+			shouldConfigureADStorage := len(c.cfg.IntrusionDetection.Spec.AnomalyDetection.StorageClassName) > 0
+
+			if !shouldConfigureADStorage {
+				objsToDelete = append(objsToDelete,
+					c.adPersistentVolumeClaim(),
+				)
+			} else if shouldConfigureADStorage && c.cfg.ShouldRenderADPVC {
+				objs = append(objs,
+					c.adPersistentVolumeClaim(),
+				)
+			}
 
-		if !shouldConfigureADStorage {
-			objsToDelete = append(objsToDelete,
-				c.adPersistentVolumeClaim(),
+			objs = append(objs,
+				c.adAPIService(),
+				c.adAPIDeployment(shouldConfigureADStorage),
 			)
-		} else if shouldConfigureADStorage && c.cfg.ShouldRenderADPVC {
+
+			// RBAC for AD Detector Pods
 			objs = append(objs,
-				c.adPersistentVolumeClaim(),
+				c.adDetectorAllowTigeraPolicy(),
+				c.adDetectorServiceAccount(),
+				c.adDetectorSecret(),
+				c.adDetectorAccessRole(),
+				c.adDetectorRoleBinding(),
 			)
+			objs = append(objs, c.adDetectorPodTemplates()...)
+		} else {
+			objsToDelete = append(objsToDelete, c.adAPIService())
+			// Always set shouldConfigureADStorage flag to false so that it won't grab the StorageClassName
+			// from IDS custom resource. Name and namespace identify the resource to be deleted.
+			objsToDelete = append(objsToDelete, c.adAPIDeployment(false))
```

Comment:
Shouldn't you delete everything else that may have been configured, like the bindings / secrets and pod templates? Is there any reason to keep them?

Reply:
The persistent volume claim should be added here too, if fips was disabled, then enabled, it wouldn't be cleaned up but you have logic above the shows cleaning it up when it's not needed should be done.

Reply:
I thought I had commented something similar that we should delete everything that may have been configured. (I'm guessing I lost it during one of my now regular Chrome crashes/kills.)